mypy/typeshed/stdlib/json/__init__.pyi

```diff
@@ -35,14 +34,13 @@ def dump(
     cls: type[JSONEncoder] | None = None,
     indent: None | int | str = None,
     separators: tuple[str, str] | None = None,
-    default: Callable[[Any], Any] | None = None,
+    default: Callable[[object], object] | None = None,
```

Comment:
callable args are contravariant, so although it looks kinda weird, I think this should be 
```suggestion
    default: Callable[[Never], object] | None = None,
```